educator/src/Dscp/Educator/Web/Educator/Queries.hs

```diff
@@ -0,0 +1,168 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+module Dscp.Educator.Web.Educator.Queries
+    ( module Dscp.Educator.Web.Educator.Queries
+    ) where
+
+import Control.Lens (from, mapping, traversed, _Just)
+import Data.List (groupBy)
+import Data.Time.Clock (getCurrentTime)
+import Database.SQLite.Simple (Only (..), Query)
+import Loot.Log (MonadLogging)
+import Servant (err501)
+import Text.InterpolatedString.Perl6 (q)
+
+import Dscp.Core
+import Dscp.Crypto
+import Dscp.DB.SQLite
+import Dscp.Educator.Web.Educator.Types
+import Dscp.Educator.Web.Types
+import Dscp.Util
+
+type MonadEducatorAPIQuery m =
+    ( MonadSQLiteDB m
+    , MonadCatch m
+    , MonadLogging m
+    )
+
+educatorRemoveStudent :: MonadEducatorAPIQuery m => Student -> m ()
+educatorRemoveStudent student = do
+    -- TODO [DSCP-176]: Proper implementation of this method may require
+    -- fundemental rethinking of our database scheme and rewriting many code.
+    -- Should be done soon though.
+    _ <- throwM err501
+    execute queryText (Only student)
+  where
+    queryText :: Query
+    queryText = [q|
+        delete
+        from      Students
+        where     addr = ?
+    |]
+
+educatorGetStudents
+    :: MonadEducatorAPIQuery m
+    => Maybe Course -> m [StudentInfo]
+educatorGetStudents courseF = do
+    map (StudentInfo . fromOnly) <$> query queryText paramF
+  where
+    (clauseF, paramF) = mkFilter "course_id = ?" courseF
+
+    queryText = [q|
+        select    addr
+        from      Students
+        left join StudentCourses
+               on Students.addr = StudentCourses.student_addr
+        where     1 = 1
+    |]
+      `filterClauses` [clauseF]
+
+educatorGetCourses :: DBM m => Maybe Student -> m [CourseEducatorInfo]
+educatorGetCourses studentF = do
+    res :: [(Course, Text, Maybe Subject)] <- query queryText paramF
+    return $
+        -- group "subject" fields for the same course
+        [ CourseEducatorInfo{..}
```

Comment:
Oh, plz no. I had to stop and undertstand why did you declared `ciSubjects` and didn't use it. You can use `NamedFieldPuns`, like that:
```
CourseEducatorInfo { ciId, ciDesc, ciSubjects }
```